static/js/components/CentroidPlot.jsx

```diff
@@ -0,0 +1,341 @@
+        fillOpacity: { value: 1.0 },
+        strokeOpacity: { value: 0 },
+      },
+    },
+
+    // Render center point (nearest object position relative to mode of the
+    // nearest references) - currently just the one reference
+    {
+      data: {
+        values: inputData.centerPoint,
+      },
+      mark: {
+        type: "point",
+        shape: "cross",
+        size: "100",
+      },
+      encoding: {
+        x: {
+          field: "delRA",
+          type: "quantitative",
+        },
+        y: {
+          field: "delDec",
+          type: "quantitative",
+        },
+        fill: { value: "black" },
+      },
+    },
+
+    // Render text messages
+    {
+      data: {
+        values: inputData.messages,
+      },
+      mark: {
+        type: "text",
+        fontSize: 14,
+        fontWeight: 500,
+      },
+      encoding: {
+        text: { field: "message", type: "nominal" },
+        x: {
+          field: "x",
+          type: "quantitative",
+        },
+        y: {
+          field: "y",
+          type: "quantitative",
+        },
+      },
+    },
+  ],
+});
+
+const processData = (photometry) => {
+  // Only take points with a non-null RA and Dec
+  const filteredPhotometry = photometry.filter(
+    (point) => point.ra && point.dec
+  );
+
+  if (filteredPhotometry.length === 0) {
+    return {
+      photometryData: [],
+    };
+  }
+
+  const ras = Object.values(filteredPhotometry).map((point) => point.ra);
+  const decs = Object.values(filteredPhotometry).map((point) => point.dec);
+
+  // For now, set single reference nearest object to median values for the RA
+  // and Dec in the photometry
+  const { refRA, refDec } = getReferencePoint(ras, decs);
+
+  const computeDeltas = (delRaGroup, delDecGroup) => (point) => {
+    const { delRA, delDec } = relativeCoord(point.ra, point.dec, refRA, refDec);
+    delRaGroup.push(delRA);
+    delDecGroup.push(delDec);
+    return {
+      ...point,
+      delRA,
+      delDec,
+    };
+  };
+
+  const delRaGroup = [];
+  const delDecGroup = [];
+  const photometryAsArray = Object.values(filteredPhotometry).map(
+    computeDeltas(delRaGroup, delDecGroup)
+  );
+
+  // Sigma circle
+  const circlePoints = getCirclePoints(delRaGroup, delDecGroup);
+
+  // Text notifications
+  const messages = getMessages(delRaGroup, delDecGroup);
+
+  const centerPoint = relativeCoord(refRA, refDec, refRA, refDec);
+
+  return {
+    photometryData: photometryAsArray,
+    circlePoints,
+    centerPoint,
+    messages,
+  };
+};
+
+const CentroidPlot = (props) => {
+  const { sourceId } = props;
+  const dispatch = useDispatch();
+  const photometry = useSelector((state) => state.photometry[sourceId]);
+
+  useEffect(() => {
+    if (!photometry) {
+      dispatch(photometryActions.fetchSourcePhotometry(sourceId));
+    }
+  }, [sourceId, photometry, dispatch]);
+
+  const plotData = photometry ? processData(photometry) : null;
+
+  let returnComponent = null;
```

Comment:
Why not just put return statements inside the blocks below and remove this variable altogether? This seems to add unnecessary complexity to me

Reply:
You're right. I think I did it initially because ESLint was complaining about returning things from else blocks or something like that, but I just managed to get rid of the variable without having ESLint yell at me with the last commit. Also included your other two suggestions.